source/includes/read/retrieve.rb

```diff
@@ -0,0 +1,47 @@
+require 'bundler/inline'
+
+gemfile do
+  source 'https://rubygems.org'
+  gem 'mongo'
+end
+
+uri = '<connection string>'
+
+begin
+  client = Mongo::Client.new(uri)
+
+  # start-db-coll
+  database = client.use('sample_training')
+  collection = database[:companies]
+  # end-db-coll
+
+  # Finds one document with a "name" value of "LinkedIn"
+  # start-find-one
+  document = collection.find({ 'name' => 'LinkedIn' }).first
+  puts document
+  # end-find-one
+
+  # Finds documents with a "founded_year" value of 1970
+  # start-find-many
+  results = collection.find({ 'founded_year' => 1970 })
+  # end-find-many
+
+  # start-cursor
+  results.each do |doc|
+    puts doc
+  end
+  # end-cursor
+
+
+  # Finds and prints up to 2 documents with a "number_of_employees" value of 1000
+  # start-modify
+  limit_results = collection.find({ 'number_of_employees' => 1000 }).limit(2)
+
+  limit_results.each do |doc|
+    puts doc
+  end
+  # end-modify
+ensure
```

Comment:
Wrapping this all in a `begin/ensure/end` block isn't wrong, but the initializer for `Mongo::Client` will do it all for you if you pass a block to it, e.g.:

```ruby
Mongo::Client.new(uri) do |client|
  # processing here, with the `client` reference.
  # the client will be closed automatically when the block terminates.
end
```

This idiom is preferred for a lot of things in Ruby, where resources need to be cleaned up after being used, like with `File.open` and network operations.